bluetooth_mesh/application.py

```diff
@@ -419,7 +419,7 @@ class Application(
 
     ELEMENTS = {}  # type: Dict[int, Type[Element]]
 
-    def __init__(self, loop: asyncio.AbstractEventLoop):
+    def __init__(self, loop: asyncio.AbstractEventLoop, token_ring: Optional[TokenRing] = None):
```

Comment:
How about making this a mixin?

```
class TokenRingMixin:
    TOKEN_RING = TokenRing

    @property
    def token_ring(self):
        reutn self.TOKEN_RING()
```

then you can just change the class when deriving from `Application`

Reply:
My inspiration was http://ccbv.co.uk/, from my XP it works wonders.

Reply:
See the current version.